### PR TITLE
[Dataset] Add peptides functional dataset in LRGB

### PR DESCRIPTION
**Pull request by paoxiaode**
## Description
The peptides functional dataset based on DGLDataset
Related issue: https://github.com/dmlc/dgl/issues/5352

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Comment by frozenbugs** [comment]
@dgl-bot

**Review by frozenbugs — APPROVED** [review]

**Comment by frozenbugs** [comment]
@rudongyu please take another look about the dataset itself.

**Review by rudongyu — APPROVED** [review]
LGTM except for a minor naming issue. Good Work！

File: python/dgl/data/lrgb.py [hunk 1]
```diff
@@ -242,3 +242,211 @@ def __getitem__(self, idx):
...
+        self,
+        raw_dir=None,
+        force_reload=None,
+        verbose=None,
+        transform=None,
+        smiles2graph=smiles2graph,
+    ):
+        self.smiles2graph = smiles2graph
+        # MD5 hash of the dataset file.
+        self.version = "701eb743e899f4d793f0e13c8fa5a1b4"
```
[the remark this answers is not in the document]
Comment (paoxiaode) [reply python/dgl/data/lrgb.py]: fixed

**Pull request merged**